elfpy/data/postgres.py

```diff
@@ -302,6 +339,46 @@ def get_pool_info(session: Session, start_block: int | None = None, end_block: i
     return pd.read_sql(query.statement, con=session.connection()).set_index("blockNumber")
 
 
+def get_checkpoint_info(session: Session, start_block: int | None = None, end_block: int | None = None) -> pd.DataFrame:
+    """Gets all info associated with a given checkpoint.
+
+    This includes
+    - `sharePrice` : The share price of the first transaction in the checkpoint.
+    - `longSharePrice` : The weighted average of the share prices that all longs in the checkpoint were opened at.
+    - `shortBaseVolume` : The aggregate amount of base committed by LPs to pay for bonds sold short in the checkpoint.
+
+    Arguments
+    ---------
+    session : Session
+        The initialized session object
+    block : int | None, optional
+        The block number whose checkpoint to return. If None, returns the most recent checkpoint.
+
+    Returns
+    -------
+    DataFrame
+        A DataFrame that consists of the queried checkpoint info
+    """
+
+    query = session.query(CheckpointInfo)
+
+    # Support for negative indices
```

Comment:
yikes, we need this?  blocks should never be negative!

Reply:
Once again, I can't speak to checkpointing, but the idea from other functions allowing negative indices was to match python list notation for accessing lists. E.g., [0, 1, 2, 3] [-1] = 3. The immediate use case is that, e.g., I want only the last 1000 blocks from the db. I can do this by passing in -1000 for the start block.

Reply:
gotcha, thanks for explanation